ecosystem/sep-0010.md

```diff
@@ -22,18 +22,23 @@ The authentication flow is as follows:
 
 1. The client obtains a unique [`challenge`](#challenge), which is represented as specially formed Stellar transaction
 1. The client verifies that the transaction has an invalid sequence number 0.  This is extremely important to ensure the transaction isn't malicious.
-1. The client signs the transaction using the secret key for the user's Stellar account
+1. The client signs the transaction using the secret key(s) of signers for the user's Stellar account
 1. The client submits the signed challenge back to the server using [`token`](#token) endpoint
+1. The server gets the signers of the user's account
+1. The server verifies the signatures
+1. The server verifies the weight provided by the signers meets the required threshold(s), if any
 1. If the signature checks out, the server responds with a [JWT](https://jwt.io) that represents the user's session
 1. Any future calls to the server can be authenticated by including the JWT as a parameter
 
 The flow achieves several things:
 
 * Both client and server part can be implemented using well-established Stellar libraries
 * The client can verify that the server holds the secret key to a particular account
-* The server can verify that the client holds the secret key to their account
-* The client is able to prove their identity using a Ledger or other hardware wallet as well as by having direct access to the secret key
+* The server can verify that the client holds the secret key(s) to signers of their account
+* The client is able to prove their identity using a Ledger or other hardware wallet as well as by having direct access to the secret key(s)
 * The server can choose its own timeout for the user's session
+* The server can choose required threshold(s) that the user needs on the account, if any
```

Comment:
Why would an authentication server choose a threshold? It doesn't know what the token will be used for. In my view of things, these servers produce tokens that other servers use for authorization (based on their own rules).

It seems like this SEP is intending for every application server to be producing its own tokens based on its own authorization rule set. I don't agree with that approach, it seems like a single authentication server producing tokens (with claims) for all application servers to use, is the correct approach.

Reply:
This SEP is leaving that in the implementers control. A server can implement a verification of required thresholds to meet its definition of a user "holding an account", and a server can add any additional claims to the JWT if it wishes to provide authentication for multiple other services where varying levels of threshold are required.